nano/node/network.cpp

```diff
@@ -468,6 +503,12 @@ class network_message_visitor : public nano::message_visitor
 
 void nano::network::process_message (nano::message const & message, std::shared_ptr<nano::transport::channel> const & channel)
 {
+	if (is_ip_blocked (channel->get_tcp_endpoint ().address ()))
```

Comment:
We shouldn't be tracking a channel for a blocked ip at all.

I think a better place to handle this is in tcp_listener::on_connection within the asinc_accept callback. The first check can be if it's a blocked IP and if so, it can issue a new call to on_connection which will wait for the next connection.